capsules/core/src/process_console.rs

```diff
@@ -1220,6 +1220,14 @@ impl<'a, const COMMAND_HISTORY_LEN: usize, A: Alarm<'a>, C: ProcessManagementCap
 
                         let cursor = self.cursor.get();
 
+                        // If we get the ANSI "DELETE" byte, we want to treat it the same as
+                        // the "delete" escape sequence. This is a dirty short-circuit.
+                        if read_buf[0] == DEL {
+                            if let EscState::Bypass = esc_state {
+                                esc_state = EscState::Complete(EscKey::Delete);
+                            }
+                        }
+
```

Comment:
It seems like this logic should just go inside the `next_state()` function for EscState.